### DISCUSSION TITLE
IM Eric Rosen vs  Fairy-Stockfish in DuckChess

### DISCUSSION THREAD

**Discussion by Vinvin20** [fairy-stockfish/Fairy-Stockfish#592] [General]
https://www.youtube.com/watch?v=Xeil4C9rU34

There's probably some weakness to find in the IA play ...

Especially this big blunder : https://youtu.be/Xeil4C9rU34?t=1248

**Comment by ianfab** [comment]
Anything but level 8 can not be taken seriously from a development perspective since it likely are intentional mistakes due to the artificial weakening.

Nevertheless, even at max strength it still is not very strong, which I am well aware of as it didn't surpass my level yet. The main challenge is to improve the move ordering and pruning decisions in order to help it cope with the huge branching factor. Furthermore, evaluation currently still is identical so chess, so not even the known big differences in piece values are reflected yet.

**Comment by defrag1971** [comment]
I give an example of weak move.
In this position Stockfish thinks best is Qb2@b1 threatening a Rook and the King (or Qa3@a2 grabbing a pawn), but as soon White replies correctly with Ra2@c1 Black eventually is forced to retreat, and FSF realizes the change in evaluation from +3 to negative values on the very next move.
Blunders like these cost FSF to lose many games.

**r1b*2nr/ppNn1k2/2p2b1p/3pp3/1q5P/P1NQB3/2P1PPP1/R3KB1R b KQ - 0 16**
```
info string classical evaluation enabled
1 106 2 1687 1 58172 0   b4a5,a5b3
2 446 5 3670 2 70576 0   b4a5,a5b6 c7a8,a8f4
3 327 9 6736 3 70905 0   b4b2,b2b1 c7a8,a8d1 b2a1,a1c4
4 504 12 9114 4 72912 0  e5e4,e4b1 a3b4,b4e5
5 371 23 18860 5 81645 0         b4b2,b2b1 c7a8,a8d1 b2a1,a1d4 c3d1,d1c5
6 357 27 22217 6 81083 0         b4b2,b2b1 c7a8,a8d1 b2a1,a1d4 c3d1,d1c5 b7b6,b6c7 d3b3,b3b7
7 365 32 25722 7 78182 0         b4b2,b2b1 c7a8,a8d1 b2a1,a1d4 c3d1,d1c5 b7b6,b6c7 d3b3,b3b7 g8e7,e7c7 h1h2,h2d4
8 395 38 28735 11 74250 0        b4b2,b2b1 c7a8,a8d1 b2a1,a1d4 c3d1,d1c5 b7b6,b6c7 e1d2,d2b4
9 361 82 57498 11 70119 0        b4b2,b2b1 c7a8,a8d1 b2a1,a1d4 c3d1,d1c5 b7b6,b6c7 d3b3,b3b7 g8e7,e7c7
10 438 146 89609 15 61166 0      b4b2,b2b1 c3d1,d1d4 b2a1,a1b2 c7a8,a8d2 d7c5,c5d4 d3d2,d2e4
11 443 314 169685 14 54022 0     b4b2,b2b1
11 417 321 173485 14 54011 0     b4b2,b2b1 c3d1,d1d4
11 392 388 206309 14 53131 0     b4b2,b2b1 c3d1,d1d4
11 360 440 236724 14 53764 0     b4b2,b2b1 c3d1,d1d4
11 389 503 267756 14 53157 0     b4b2,b2b1
11 316 550 289007 14 52460 0     b4b2,b2b1 c3d1,d1d4
11 385 576 302372 15 52476 0     b4b2,b2b1
11 346 658 342809 15 52090 0     b4b2,b2b1 c7a8,a8d1 b2a1,a1d4 c3d1,d1c5 a1a3,a3c3 a8c7,c7e7 d5d4,d4c3
12 337 997 543804 14 54527 0     b4b2,b2b1 c3d1,d1d4
12 329 1170 644787 14 55067 0    b4b2,b2b1 c3d1,d1d4
12 317 1286 715136 14 55570 0    b4b2,b2b1 c3d1,d1d4
12 327 1338 744721 14 55650 0    b4b2,b2b1
12 351 1365 762244 14 55829 0    b4b2,b2b1
12 384 1397 785031 14 56194 0    b4b2,b2b1
12 297 1409 792081 14 56215 0    b4b2,b2b1 c3d1,d1d4
12 363 1560 889955 15 57026 0    b4b2,b2b1
12 346 1600 915239 15 57202 0    b4b2,b2b1 c3d1,d1d4 b2a1,a1b2 c7a8,a8e4 a1a3,a3c3 g2g4,g4e4 g8e7,e7b3
13 337 1852 1074478 16 58007 0   b4b2,b2b1 c3d1,d1d4
13 329 1917 1111095 16 57954 0   b4b2,b2b1 c3d1,d1d4
```

after Ra2@c1
**r1b3nr/ppNn1k2/2p2b1p/3pp3/7P/P1NQB3/RqP1PPP1/2*1KB1R b K - 2 17**
```
info string classical evaluation enabled
1 107 2 1827 1 73080 0   e5e4,e4a1 a2b2,b2d1
2 107 6 3339 2 55650 0   e5e4,e4a1 a2b2,b2d1
3 146 9 5245 3 57637 0   e5e4,e4a1 a2b2,b2d1 e4d3,d3h3
4 -146 25 19856 4 76664 0        e5e4,e4e5 a2b2,b2d1 e4d3,d3e5 e2d3,d3d4
5 -43 28 21618 5 76119 0         e5e4,e4e5 a2b2,b2d1 e4d3,d3d2 c2d3,d3e5
6 30 40 27398 6 68154 0  b2c3,c3d4 d3c3,c3d1 d5d4,d4d2
7 133 43 30245 7 69052 0         b2c3,c3d4 d3c3,c3d1 d5d4,d4d2
8 614 48 33858 8 69666 0         b2c3,c3d4 d3c3,c3d1 d5d4,d4d2 c7a8,a8e4
9 -258 172 110823 11 64096 0     e5e4,e4e5 a2b2,b2d4 e4d3,d3e5 c7a8,a8d4 d3e2,e2d1 f1e2,e2d4 f7g7,g7d1
10 -250 288 175421 11 60762 0    e5e4,e4e5 a2b2,b2d4 e4d3,d3e5 c7a8,a8d4 d3c2,c2e5 b2c2,c2d4 b7b6,b6c4
11 -266 374 237659 14 63511 0    e5e4,e4e5 a2b2,b2d4
11 -224 399 262525 14 65696 0    e5e4,e4e5
11 -189 409 271950 14 66475 0    e5e4,e4e5
ex11 -146 458 303718 14 66299 0  e5e4,e4e5
i11 -284 489 317614 14 64858 0   e5e4,e4e5 a2b2,b2d4
t11 -181 497 327209 14 65823 0   e5e4,e4e5
11 -181 583 381072 14 65363 0    e5e4,e4e5
```

**Comment by Vinvin20** [comment]
There's a strong opponent if you want to test FSF in real games : http://talkchess.com/forum3/viewtopic.php?p=939064#p939064

**Comment by davidspencer6174** [comment]
I wonder whether there's some bug. Most of my games against it ended with it missing king-capture in 2 (basically mate in 1) in situations where it could have survived longer. Some examples are [this](https://www.pychess.org/GP5xtFl7) and [this](https://www.pychess.org/YKxTsjbt) . Someone else beat it in [7 moves](https://www.pychess.org/IkHVEVRy)

Maybe it's just because of extremely low depth/thinking time on the server, though. I saw the [thread on talkchess](https://talkchess.com/forum3/viewtopic.php?f=7&t=80829&sid=6ab86f6a0dacd5c197b0a4eb6c5d94ed&start=100#p939176) where it holds its own against the TJDuckChess engine in some time controls.

Anyway, thanks for developing this! Hope to see it grow stronger and stronger.

**Comment by ianfab** [comment]
Missing checkmates quite likely has to do with the way the rules are implemented. Since the kings are not treated as royal pieces it does not recognize check and checkmate, but only recognizes a position as losing when the king is captured. Hence it needs to effectively search 3 ply for a mate in one (mate + random move + capture king), which can take a while considering the large branching factor. There very likely is a lot of room for improvement there.
